Crash when Load Balancer Custom Rules create with pop_pools or region_pools

### Confirmation

My issue isn't already found on the issue tracker.
I have replicated my issue using the latest version of the provider and it is still present.

### Terraform version

Bug reproducible with 0.13.6, 0.14.0, 0.15.0
And with `terraform apply -parallelism=1`


### Affected resource(s)

Custom Rules with pop_pools and region_pools

### Terraform configuration files

```hcl
resource "cloudflare_load_balancer" "main" {
  for_each = var.cloudflare_load_balancer

  name    = each.value["name"]
  zone_id = var.cloudflare_zone_id

  fallback_pool_id = local.pools[each.value["fallback_pool_id"]]
  default_pool_ids = [for default_pool_ids in each.value["default_pool_ids"] : local.pools[default_pool_ids]]

  dynamic rules {
    for_each = [for rules in lookup(each.value, "rules", null) : rules]
    iterator = rules

    content {
      name = lookup(rules.value, "name", null)

      dynamic overrides {
        for_each = [for overrides in lookup(rules.value, "overrides", {}): overrides]
        iterator = overrides

        content {
          session_affinity = lookup(overrides.value, "session_affinity", null)
          session_affinity_ttl = lookup(overrides.value, "session_affinity_ttl", null)
          session_affinity_attributes = lookup(overrides.value, "session_affinity_attributes", null)
          ttl = lookup(overrides.value, "ttl", null)
          steering_policy = lookup(overrides.value, "steering_policy", null)
          fallback_pool   = lookup(overrides.value, "fallback_pool", null)
          default_pools   = [for default_pool in lookup(overrides.value, "default_pools", []) : local.pools[default_pool]]

          dynamic region_pools {
            for_each = [for region_pool in lookup(overrides.value, "region_pools", {}): region_pool]
            iterator = region_pool

            content {
              region = lookup(region_pool.value, "region", null)
              pool_ids = [for pool_id in lookup(region_pool.value, "pool_ids", []) : local.pools[pool_id]]
            }
          }
        }

          dynamic pop_pools {
            for_each = [for pop_p in lookup(overrides.value, "pop_pools", {}): pop_p]
            iterator = pop_p

            content {
              pop = lookup(pop_p.value, "pop", null)
              pool_ids = [for pool_id in lookup(pop_p.value, "pool_ids", []) : local.pools[pool_id]]
            }
          }
        }
      }
    }
  }

  session_affinity            = lookup(each.value, "session_affinity", "none")
  session_affinity_ttl        = lookup(each.value, "session_affinity_ttl", 82800)
  session_affinity_attributes = lookup(each.value, "session_affinity_attributes", {})
}
```


### Debug output


  # module.cloudflare_load_balancer["dev"].cloudflare_load_balancer.main["test_lb1"] will be created
  + resource "cloudflare_load_balancer" "main" {
      + created_on           = (known after apply)
      + default_pool_ids     = [
          + "bd7c023b5a0868bd28755e358a829802",
        ]
      + description          = "test_load_balancer"
      + enabled              = true
      + fallback_pool_id     = "bd7c023b5a0868bd28755e358a829802"
      + id                   = (known after apply)
      + modified_on          = (known after apply)
      + name                 = "test_lb1"
      + proxied              = true
      + session_affinity     = "none"
      + session_affinity_ttl = 82800
      + steering_policy      = (known after apply)
      + ttl                  = (known after apply)
      + zone_id              = "a13b50fe516ee59ed462c1fdf56cb3ca"

      + pop_pools {
          + pool_ids = (known after apply)
          + pop      = (known after apply)
        }

      + region_pools {
          + pool_ids = (known after apply)
          + region   = (known after apply)
        }

      + rules {
          + condition  = "(http.request.uri.path contains \"/test\")"
          + disabled   = false
          + name       = "test rule 3"
          + priority   = (known after apply)
          + terminates = false

          + overrides {
              + default_pools   = []
              + steering_policy = "geo"

              + pop_pools {
                  + pool_ids = [
                      + "af0e7fb45dcfc3e0e2a5bcfa3d0a8920",
                    ]
                  + pop      = "LAX"
                }
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

Enter a value: 2021-04-15T19:17:34.762+0300 [DEBUG] command: asking for input: "\nDo you want to perform these actions?"
yes

2021-04-15T19:17:37.898+0300 [INFO]  backend/local: apply calling Apply
2021-04-15T19:17:37.898+0300 [INFO]  terraform: building graph: GraphTypeApply
2021-04-15T19:17:37.901+0300 [DEBUG] Resource state not found for node "module.cloudflare_load_balancer[\"dev\"].cloudflare_load_balancer.main[\"test_lb1\"]", instance module.cloudflare_load_balancer["dev"].cloudflare_load_balancer.main["test_lb1"]
2021-04-15T19:17:37.901+0300 [DEBUG] adding implicit provider configuration provider["registry.terraform.io/cloudflare/cloudflare"], implied first by module.cloudflare_load_balancer.cloudflare_load_balancer_pool.main (expand)
2021-04-15T19:17:37.901+0300 [DEBUG] ProviderTransformer: "module.cloudflare_load_balancer[\"dev\"].cloudflare_load_balancer.main[\"test_lb1\"]" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2021-04-15T19:17:37.901+0300 [DEBUG] ProviderTransformer: "module.cloudflare_load_balancer.cloudflare_load_balancer_pool.main (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2021-04-15T19:17:37.902+0300 [DEBUG] ProviderTransformer: "module.cloudflare_load_balancer.cloudflare_load_balancer.main (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2021-04-15T19:17:37.902+0300 [DEBUG] ProviderTransformer: "module.cloudflare_load_balancer.cloudflare_load_balancer_monitor.main (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" references: []
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "var.cloudflare_zone_id" references: []
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "var.cloudflare_load_balancer_monitor" references: []
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.var.test (expand)" references: []
2021-04-15T19:17:37.903+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.var.cloudflare_zone_id (expand)" references: []
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "local.cloudflare_load_balancer (expand)" references: [local.zone_cloudflare_load_balancer (expand) var.cloudflare_load_balancer local.zone_cloudflare_load_balancer (expand)]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "local.zone_cloudflare_load_balancer_pool (expand)" references: [var.cloudflare_zone_id var.cloudflare_load_balancer_pool var.cloudflare_load_balancer_pool]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "local.cloudflare_load_balancer_pool (expand)" references: [local.zone_cloudflare_load_balancer_pool (expand) var.cloudflare_load_balancer_pool local.zone_cloudflare_load_balancer_pool (expand)]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "local.zone_cloudflare_load_balancer (expand)" references: [var.cloudflare_zone_id var.cloudflare_load_balancer var.cloudflare_load_balancer]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.cloudflare_load_balancer.main (expand)" references: [module.cloudflare_load_balancer.var.cloudflare_load_balancer (expand)]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "var.cloudflare_load_balancer_pool" references: []
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "var.cloudflare_load_balancer" references: []
2021-04-15T19:17:37.903+0300 [INFO]  ReferenceTransformer: reference not found: "each.key"
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.var.cloudflare_load_balancer_pool (expand)" references: [local.cloudflare_load_balancer_pool (expand)]
2021-04-15T19:17:37.903+0300 [INFO]  ReferenceTransformer: reference not found: "each.key"
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.var.cloudflare_load_balancer (expand)" references: [local.cloudflare_load_balancer (expand)]
2021-04-15T19:17:37.903+0300 [INFO]  ReferenceTransformer: reference not found: "each.key"
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.var.cloudflare_load_balancer_monitor (expand)" references: [local.cloudflare_load_balancer_monitor (expand)]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "local.zone_cloudflare_load_balancer_monitor (expand)" references: [var.cloudflare_zone_id var.cloudflare_load_balancer_monitor var.cloudflare_load_balancer_monitor]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer (close)" references: []
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.cloudflare_load_balancer_monitor.main (expand)" references: [module.cloudflare_load_balancer.var.cloudflare_load_balancer_monitor (expand)]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "local.cloudflare_load_balancer_monitor (expand)" references: [local.zone_cloudflare_load_balancer_monitor (expand) var.cloudflare_load_balancer_monitor local.zone_cloudflare_load_balancer_monitor (expand)]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.cloudflare_load_balancer_pool.main (expand)" references: [module.cloudflare_load_balancer.var.cloudflare_load_balancer_pool (expand)]
2021-04-15T19:17:37.903+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer.local.pools (expand)" references: [module.cloudflare_load_balancer.cloudflare_load_balancer_pool.main (expand)]
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [INFO]  ReferenceTransformer: reference not found: "each.value"
2021-04-15T19:17:37.904+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer[\"dev\"].cloudflare_load_balancer.main[\"test_lb1\"]" references: [module.cloudflare_load_balancer.var.cloudflare_load_balancer (expand) module.cloudflare_load_balancer.var.cloudflare_zone_id (expand) module.cloudflare_load_balancer.local.pools (expand) module.cloudflare_load_balancer.local.pools (expand) module.cloudflare_load_balancer.local.pools (expand) module.cloudflare_load_balancer.local.pools (expand) module.cloudflare_load_balancer.local.pools (expand) module.cloudflare_load_balancer.local.pools (expand)]
2021-04-15T19:17:37.904+0300 [DEBUG] ReferenceTransformer: "module.cloudflare_load_balancer (expand)" references: [var.cloudflare_zone_id]
2021-04-15T19:17:37.906+0300 [DEBUG] pruneUnusedNodes: module.cloudflare_load_balancer.cloudflare_load_balancer.main (expand) is no longer needed, removing
2021-04-15T19:17:37.906+0300 [DEBUG] pruneUnusedNodes: module.cloudflare_load_balancer.cloudflare_load_balancer_monitor.main (expand) is no longer needed, removing
2021-04-15T19:17:37.906+0300 [DEBUG] pruneUnusedNodes: module.cloudflare_load_balancer.var.test (expand) is no longer needed, removing
2021-04-15T19:17:37.906+0300 [DEBUG] pruneUnusedNodes: module.cloudflare_load_balancer.var.cloudflare_load_balancer_monitor (expand) is no longer needed, removing
2021-04-15T19:17:37.906+0300 [DEBUG] pruneUnusedNodes: local.cloudflare_load_balancer_monitor (expand) is no longer needed, removing
2021-04-15T19:17:37.906+0300 [DEBUG] pruneUnusedNodes: local.zone_cloudflare_load_balancer_monitor (expand) is no longer needed, removing
2021-04-15T19:17:37.910+0300 [DEBUG] Starting graph walk: walkApply
2021-04-15T19:17:37.911+0300 [DEBUG] created provider logger: level=debug
2021-04-15T19:17:37.911+0300 [INFO]  provider: configuring client automatic mTLS
2021-04-15T19:17:37.961+0300 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/2.20.0/darwin_amd64/terraform-provider-cloudflare_v2.20.0 args=[.terraform/providers/registry.terraform.io/cloudflare/cloudflare/2.20.0/darwin_amd64/terraform-provider-cloudflare_v2.20.0]
2021-04-15T19:17:37.969+0300 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/2.20.0/darwin_amd64/terraform-provider-cloudflare_v2.20.0 pid=12497
2021-04-15T19:17:37.970+0300 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/2.20.0/darwin_amd64/terraform-provider-cloudflare_v2.20.0
2021-04-15T19:17:37.990+0300 [INFO]  provider.terraform-provider-cloudflare_v2.20.0: configuring server automatic mTLS: timestamp=2021-04-15T19:17:37.990+0300
2021-04-15T19:17:38.033+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: plugin address: address=/var/folders/jm/nl012fkn62l7vdp94w7rwf080000gn/T/plugin696901052 network=unix timestamp=2021-04-15T19:17:38.033+0300
2021-04-15T19:17:38.034+0300 [DEBUG] provider: using plugin: version=5
2021-04-15T19:17:38.096+0300 [DEBUG] No provider meta schema returned
2021-04-15T19:17:38.105+0300 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" changed the config value, but that value is unused
2021-04-15T19:17:38.106+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:38 [INFO] Cloudflare Client configured for user: 
2021-04-15T19:17:38.106+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:38 [INFO] Using specified account id <id> in Cloudflare provider
2021-04-15T19:17:38.106+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:38 [INFO] Cloudflare Client configured for user: 
2021-04-15T19:17:38.114+0300 [WARN]  Provider "registry.terraform.io/cloudflare/cloudflare" produced an invalid plan for module.cloudflare_load_balancer["dev"].cloudflare_load_balancer.main["test_lb1"], but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .session_affinity_attributes: planned value cty.NullVal(cty.Map(cty.String)) does not match config value cty.MapValEmpty(cty.String)
      - .steering_policy: planned value cty.UnknownVal(cty.String) does not match config value cty.StringVal("")
      - .pop_pools: attribute representing nested block must not be unknown itself; set nested attribute values to unknown instead
      - .region_pools: attribute representing nested block must not be unknown itself; set nested attribute values to unknown instead
      - .rules[0].fixed_response: planned value cty.NullVal(cty.Map(cty.String)) does not match config value cty.MapValEmpty(cty.String)
module.cloudflare_load_balancer["dev"].cloudflare_load_balancer.main["test_lb1"]: Creating...
2021-04-15T19:17:38.115+0300 [INFO]  Starting apply for module.cloudflare_load_balancer["dev"].cloudflare_load_balancer.main["test_lb1"]
2021-04-15T19:17:38.117+0300 [DEBUG] module.cloudflare_load_balancer["dev"].cloudflare_load_balancer.main["test_lb1"]: applying the planned Create change
2021-04-15T19:17:38.118+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:38 [DEBUG] setting computed for "region_pools" from ComputedKeys
2021-04-15T19:17:38.119+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:38 [DEBUG] setting computed for "pop_pools" from ComputedKeys
2021-04-15T19:17:38.120+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:38 [INFO] Creating Cloudflare Load Balancer from struct: {ID: CreatedOn:<nil> ModifiedOn:<nil> Description:test_load_balancer Name:test_lb1.example.com TTL:0 FallbackPool:bd7c023b5a0868bd28755e358a829802 DefaultPools:[bd7c023b5a0868bd28755e358a829802] RegionPools:map[] PopPools:map[] Proxied:true Enabled:0xc0006b27d8 Persistence:none PersistenceTTL:82800 SessionAffinityAttributes:<nil> Rules:[0xc00066c210] SteeringPolicy:}
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:38 [DEBUG] Cloudflare API Request Details:
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: ---[ REQUEST ]---------------------------------------
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: POST /client/v4/zones/a13b50fe516ee59ed462c1fdf56cb3ca/load_balancers HTTP/1.1
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Host: api.cloudflare.com
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: User-Agent: HashiCorp Terraform/0.15.0 (+https://www.terraform.io) Terraform Plugin SDK/1.16.1 terraform-provider-cloudflare/2.20.0
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Content-Length: 513
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Authorization: Bearer
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Content-Type: application/json
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Accept-Encoding: gzip
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: {
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "description": "test_load_balancer",
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "name": "test_lb1.example.com",
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "fallback_pool": "bd7c023b5a0868bd28755e358a829802",
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "default_pools": [
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "bd7c023b5a0868bd28755e358a829802"
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  ],
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "region_pools": null,
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "pop_pools": null,
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "proxied": true,
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "enabled": true,
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "session_affinity": "none",
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "session_affinity_ttl": 82800,
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  "rules": [
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   {
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:    "name": "test rule 3",
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:    "priority": 0,
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:    "disabled": false,
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:    "condition": "(http.request.uri.path contains \"/cnnsdf\")",
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:    "overrides": {
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "steering_policy": "geo",
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "pop_pools": {
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:      "LAX": [
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       "af0e7fb45dcfc3e0e2a5bcfa3d0a8920"
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:      ]
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     }
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:    }
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   }
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:  ]
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: }
2021-04-15T19:17:38.123+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: -----------------------------------------------------
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:39 [DEBUG] Cloudflare API Response Details:
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: ---[ RESPONSE ]--------------------------------------
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: HTTP/2.0 200 OK
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Cf-Cache-Status: DYNAMIC
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Cf-Ray: 64067c36dc4a0c42-DME
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Cf-Request-Id: 0977ebf64400000c4219832000000001
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Content-Type: application/json; charset=UTF-8
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Date: Thu, 15 Apr 2021 16:17:39 GMT
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Expect-Ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Server: cloudflare
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Set-Cookie: __cfduid=d7302158f18d1617a2a7d372e5b58b1461618503458; expires=Sat, 15-May-21 16:17:38 GMT; path=/; domain=.api.cloudflare.com; HttpOnly; SameSite=Lax; Secure
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Set-Cookie: __cflb=0H28vgHxwvgAQtjUGU56Rb8iNWZVUvXhmu7R4NptVob; SameSite=Lax; path=/; expires=Thu, 15-Apr-21 18:47:40 GMT; HttpOnly
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Set-Cookie: __cfruid=7b4a6330e0971dc6c26050eb01094f6fca2d0a4b-1618503459; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Vary: Accept-Encoding
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: X-Envoy-Upstream-Service-Time: 45
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: {
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "result": {
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "description": "test_load_balancer",
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "created_on": "2021-04-15T16:17:39.248777Z",
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "modified_on": "2021-04-15T16:17:39.248777Z",
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "id": "c525cd90e42298b7c78fb79a726646b4",
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "ttl": 30,
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "proxied": true,
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "enabled": true,
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "name": "test_lb1.example.com",
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "session_affinity": "none",
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "session_affinity_attributes": {
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       "samesite": "Auto",
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       "secure": "Auto",
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       "drain_duration": 0
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     },
2021-04-15T19:17:39.413+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "steering_policy": "off",
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "fallback_pool": "bd7c023b5a0868bd28755e358a829802",
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "default_pools": [
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: "bd7c023b5a0868bd28755e358a829802"
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     ],
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "pop_pools": {},
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "region_pools": {},
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "rules": [
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       {
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "name": "test rule 3",
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "priority": 0,
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "disabled": false,
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "condition": "(http.request.uri.path contains \"/cnnsdf\")",
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "overrides": {
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:           "steering_policy": "geo",
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:           "pop_pools": {
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:             "LAX": [
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: "af0e7fb45dcfc3e0e2a5bcfa3d0a8920"
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:             ]
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:           }
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         }
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       }
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     ]
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   },
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "success": true,
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "errors": [],
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "messages": []
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: }
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: -----------------------------------------------------
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:39 [INFO] Cloudflare Load Balancer ID: c525cd90e42298b7c78fb79a726646b4
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:39 [DEBUG] Cloudflare API Request Details:
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: ---[ REQUEST ]---------------------------------------
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: GET /client/v4/zones/a13b50fe516ee59ed462c1fdf56cb3ca/load_balancers/c525cd90e42298b7c78fb79a726646b4 HTTP/1.1
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Host: api.cloudflare.com
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: User-Agent: HashiCorp Terraform/0.15.0 (+https://www.terraform.io) Terraform Plugin SDK/1.16.1 terraform-provider-cloudflare/2.20.0
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Authorization: Bearer
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Content-Type: application/json
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Accept-Encoding: gzip
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 
2021-04-15T19:17:39.414+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: -----------------------------------------------------
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 2021/04/15 19:17:40 [DEBUG] Cloudflare API Response Details:
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: ---[ RESPONSE ]--------------------------------------
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: HTTP/2.0 200 OK
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Cf-Cache-Status: DYNAMIC
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Cf-Ray: 64067c3e486efa94-AMS
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Cf-Request-Id: 0977ebfaec0000fa9443253000000001
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Content-Type: application/json; charset=UTF-8
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Date: Thu, 15 Apr 2021 16:17:40 GMT
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Expect-Ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Server: cloudflare
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Set-Cookie: __cfduid=d528dd51e87873f2d4995f8f147b1e32f1618503459; expires=Sat, 15-May-21 16:17:39 GMT; path=/; domain=.api.cloudflare.com; HttpOnly; SameSite=Lax; Secure
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Set-Cookie: __cflb=0H28vgHxwvgAQtjUGU56Rb8iNWZVUvXhn4p65BeP757; SameSite=Lax; path=/; expires=Thu, 15-Apr-21 18:47:41 GMT; HttpOnly
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Set-Cookie: __cfruid=37b9d58d2f4b9d7cb420cbebb69cf96b8fc6b08a-1618503460; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: Vary: Accept-Encoding
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: X-Envoy-Upstream-Service-Time: 10
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: {
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "result": {
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "description": "test_load_balancer",
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "created_on": "2021-04-15T16:17:39.248777Z",
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "modified_on": "2021-04-15T16:17:39.248777Z",
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "id": "c525cd90e42298b7c78fb79a726646b4",
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "proxied": true,
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "enabled": true,
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "name": "test_lb1.example.com",
2021-04-15T19:17:40.604+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "session_affinity": "none",
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "session_affinity_attributes": {
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       "samesite": "Auto",
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       "secure": "Auto",
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       "drain_duration": 0
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     },
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "steering_policy": "off",
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "fallback_pool": "bd7c023b5a0868bd28755e358a829802",
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "default_pools": [
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: "bd7c023b5a0868bd28755e358a829802"
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     ],
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "pop_pools": {},
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "region_pools": {},
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     "rules": [
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       {
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "name": "test rule 3",
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "priority": 0,
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "disabled": false,
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "condition": "(http.request.uri.path contains \"/cnnsdf\")",
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         "overrides": {
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:           "steering_policy": "geo",
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:           "pop_pools": {
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:             "LAX": [
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: "af0e7fb45dcfc3e0e2a5bcfa3d0a8920"
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:             ]
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:           }
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:         }
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:       }
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:     ]
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   },
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "success": true,
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "errors": [],
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0:   "messages": []
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: }
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 
2021-04-15T19:17:40.605+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: -----------------------------------------------------


### Panic output

2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: PANIC: runtime error: invalid memory address or nil pointer dereference
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19b9891]
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: goroutine 22 [running]:
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.SerializeResourceForHash(0xc000738880, 0x1ce8060, 0xc000352780, 0x0)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/serialize.go:93 +0x51
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.HashResource.func1(0x1ce8060, 0xc000352780, 0x1eb1708)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:32 +0x68
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Set).hash(0xc0004aa100, 0x1ce8060, 0xc000352780, 0xc0001886b0, 0xc000738990)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:254 +0x3d
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Set).add(0xc0004aa100, 0x1ce8060, 0xc000352780, 0x0, 0xc0001886c0, 0xc0007389e0)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:231 +0x85
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Set).Add(...)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:75
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.NewSet(0xc0001886c0, 0xc0001886b0, 0x1, 0x1, 0x26f70c0)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:62 +0xa5
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.flattenGeoPools(0xc000352120, 0x1e59ce0, 0x9, 0x8)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer.go:552 +0x3a5
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.flattenRules(0xc0006593b0, 0xc0004aa020, 0x1, 0x4, 0xc000352210, 0x0, 0x0, 0xc0006ba2e0)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer.go:666 +0x14e5
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareLoadBalancerRead(0xc0006593b0, 0x1e514a0, 0xc000542180, 0x1, 0x1)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer.go:515 +0xf8e
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareLoadBalancerCreate(0xc0006593b0, 0x1e514a0, 0xc000542180, 0x2, 0x26f70c0)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer.go:393 +0x970
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc00049cab0, 0xc0006486e0, 0xc0005487c0, 0x1e514a0, 0xc000542180, 0x1ce5c01, 0xc00038ea58, 0xc000393380)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/resource.go:310 +0x375
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc0005a6280, 0xc0004b9a38, 0xc0006486e0, 0xc0005487c0, 0xc0002d2e08, 0xc0000319e0, 0x1ce8060)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/provider.go:294 +0x99
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00010ec18, 0x2061a10, 0xc0006461b0, 0xc000658000, 0xc00010ec18, 0xc0006461b0, 0xc000202ba0)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/internal/helper/plugin/grpc_provider.go:885 +0x8a5
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1e11d40, 0xc00010ec18, 0x2061a10, 0xc0006461b0, 0xc000110900, 0x0, 0x2061a10, 0xc0006461b0, 0xc000656c00, 0x5f5)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003461c0, 0x206beb8, 0xc000083080, 0xc000650000, 0xc0002c24e0, 0x26b4f80, 0x0, 0x0, 0x0)
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	google.golang.org/grpc@v1.31.1/server.go:1180 +0x52b
2021-04-15T19:17:40.607+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: google.golang.org/grpc.(*Server).handleStream(0xc0003461c0, 0x206beb8, 0xc000083080, 0xc000650000, 0x0)
2021-04-15T19:17:40.608+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	google.golang.org/grpc@v1.31.1/server.go:1503 +0xd0c
2021-04-15T19:17:40.608+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000038250, 0xc0003461c0, 0x206beb8, 0xc000083080, 0xc000650000)
2021-04-15T19:17:40.608+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	google.golang.org/grpc@v1.31.1/server.go:843 +0xab
2021-04-15T19:17:40.608+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-04-15T19:17:40.608+0300 [DEBUG] provider.terraform-provider-cloudflare_v2.20.0: 	google.golang.org/grpc@v1.31.1/server.go:841 +0x1fd
2021-04-15T19:17:40.610+0300 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-04-15T19:17:40.610+0300 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/2.20.0/darwin_amd64/terraform-provider-cloudflare_v2.20.0 pid=12497 error="exit status 2"
2021-04-15T19:17:40.619+0300 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = transport is closing"
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-cloudflare_v2.20.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19b9891]

goroutine 22 [running]:
github.com/hashicorp/terraform-plugin-sdk/helper/schema.SerializeResourceForHash(0xc000738880, 0x1ce8060, 0xc000352780, 0x0)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/serialize.go:93 +0x51
github.com/hashicorp/terraform-plugin-sdk/helper/schema.HashResource.func1(0x1ce8060, 0xc000352780, 0x1eb1708)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:32 +0x68
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Set).hash(0xc0004aa100, 0x1ce8060, 0xc000352780, 0xc0001886b0, 0xc000738990)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:254 +0x3d
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Set).add(0xc0004aa100, 0x1ce8060, 0xc000352780, 0x0, 0xc0001886c0, 0xc0007389e0)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:231 +0x85
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Set).Add(...)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:75
github.com/hashicorp/terraform-plugin-sdk/helper/schema.NewSet(0xc0001886c0, 0xc0001886b0, 0x1, 0x1, 0x26f70c0)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/set.go:62 +0xa5
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.flattenGeoPools(0xc000352120, 0x1e59ce0, 0x9, 0x8)
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer.go:552 +0x3a5
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.flattenRules(0xc0006593b0, 0xc0004aa020, 0x1, 0x4, 0xc000352210, 0x0, 0x0, 0xc0006ba2e0)
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer.go:666 +0x14e5
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareLoadBalancerRead(0xc0006593b0, 0x1e514a0, 0xc000542180, 0x1, 0x1)
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer.go:515 +0xf8e
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareLoadBalancerCreate(0xc0006593b0, 0x1e514a0, 0xc000542180, 0x2, 0x26f70c0)
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer.go:393 +0x970
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc00049cab0, 0xc0006486e0, 0xc0005487c0, 0x1e514a0, 0xc000542180, 0x1ce5c01, 0xc00038ea58, 0xc000393380)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/resource.go:310 +0x375
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc0005a6280, 0xc0004b9a38, 0xc0006486e0, 0xc0005487c0, 0xc0002d2e08, 0xc0000319e0, 0x1ce8060)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00010ec18, 0x2061a10, 0xc0006461b0, 0xc000658000, 0xc00010ec18, 0xc0006461b0, 0xc000202ba0)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1e11d40, 0xc00010ec18, 0x2061a10, 0xc0006461b0, 0xc000110900, 0x0, 0x2061a10, 0xc0006461b0, 0xc000656c00, 0x5f5)
	github.com/hashicorp/terraform-plugin-sdk@v1.16.1/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003461c0, 0x206beb8, 0xc000083080, 0xc000650000, 0xc0002c24e0, 0x26b4f80, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.31.1/server.go:1180 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0003461c0, 0x206beb8, 0xc000083080, 0xc000650000, 0x0)
	google.golang.org/grpc@v1.31.1/server.go:1503 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000038250, 0xc0003461c0, 0x206beb8, 0xc000083080, 0xc000650000)
	google.golang.org/grpc@v1.31.1/server.go:843 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.31.1/server.go:841 +0x1fd

Error: The terraform-provider-cloudflare_v2.20.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2021-04-15T19:17:40.644+0300 [DEBUG] provider: plugin exited

### Expected output

Load Balancer created with proper Custom Rules and pop_pools without terraform crash

### Actual output

Load Balancer created with proper Custom Rules and pop_pools with terraform crash

### Steps to reproduce

terraform init
terraform apply

### Additional factoids

Load balancer with custom rules and with pop_pools (and region_pools as well) created successfully, and in the logs, we see that Cloudflare respond to us:
```
"success": true,
"errors": [],
"messages": []
```

Also, custom rules with other overrides or fixed_response rules created successfully.

### References

_No response_

--- a/cloudflare/resource_cloudflare_load_balancer.go
+++ b/cloudflare/resource_cloudflare_load_balancer.go
@@ -663,11 +663,11 @@ func flattenRules(d *schema.ResourceData, rules []*cloudflare.LoadBalancerRule) 
 				m["overrides"] = []interface{}{om}
 			}
 			if _, ok := d.GetOkExists(fmt.Sprintf("rules.%d.overrides.0.pop_pools", idx)); ok {
-				om["pop_pools"] = flattenGeoPools(o.PoPPools, "rules_pop")
+				om["pop_pools"] = flattenGeoPools(o.PoPPools, "pop")
 				m["overrides"] = []interface{}{om}
 			}
 			if _, ok := d.GetOkExists(fmt.Sprintf("rules.%d.overrides.0.region_pools", idx)); ok {
-				om["region_pools"] = flattenGeoPools(o.RegionPools, "rules_region")
+				om["region_pools"] = flattenGeoPools(o.RegionPools, "region")
 				m["overrides"] = []interface{}{om}
 			}
 			if _, ok := d.GetOkExists(fmt.Sprintf("rules.%d.overrides.0.session_affinity_attributes", idx)); o.SessionAffinityAttrs != nil && ok {

--- a/cloudflare/resource_cloudflare_load_balancer_test.go
+++ b/cloudflare/resource_cloudflare_load_balancer_test.go
@@ -166,8 +166,9 @@ func TestAccCloudflareLoadBalancer_Rules(t *testing.T) {
 					resource.TestCheckResourceAttr(name, "rules.0.overrides.0.steering_policy", "geo"),
 					resource.TestCheckResourceAttr(name, "rules.0.overrides.0.session_affinity_attributes.samesite", "Auto"),
 					resource.TestCheckResourceAttr(name, "rules.0.overrides.0.session_affinity_attributes.secure", "Auto"),
-					resource.TestCheckResourceAttr(name, "rules.#", "2"),
+					resource.TestCheckResourceAttr(name, "rules.#", "3"),
 					resource.TestCheckResourceAttr(name, "rules.1.fixed_response.message_body", "hello"),
+					resource.TestCheckResourceAttr(name, "rules.2.overrides.0.region_pools.0.region", "ENAM"),
 				),
 			},
 		},
@@ -491,6 +492,16 @@ resource "cloudflare_load_balancer" "%[3]s" {
       status_code = "200"
       content_type = "html"
       location = "www.example.com"
+    }
+  }
+  rules {
+    name = "test rule 3"
+    condition = "dns.qry.type == 28"
+    overrides {
+      region_pools {
+		    region = "ENAM"
+		    pool_ids = ["${cloudflare_load_balancer_pool.%[3]s.id}"]
+	    }
     }
   }
 }`, zoneID, zone, id)

--- a/cloudflare/resource_cloudflare_load_balancer_test.go
+++ b/cloudflare/resource_cloudflare_load_balancer_test.go
@@ -168,7 +168,7 @@ func TestAccCloudflareLoadBalancer_Rules(t *testing.T) {
 					resource.TestCheckResourceAttr(name, "rules.0.overrides.0.session_affinity_attributes.secure", "Auto"),
 					resource.TestCheckResourceAttr(name, "rules.#", "3"),
 					resource.TestCheckResourceAttr(name, "rules.1.fixed_response.message_body", "hello"),
-					resource.TestCheckResourceAttr(name, "rules.2.overrides.0.region_pools.0.region", "ENAM"),
+					resource.TestCheckResourceAttr(name, "rules.2.overrides.0.region_pools.#", "1"),
 				),
 			},
 		},

